internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1073,3 +1076,45 @@ func (r *ClusterDeploymentsReconciler) ensureOwnerRef(ctx context.Context, cd *h
 	}
 	return r.Update(ctx, ci)
 }
+
+func (r *ClusterDeploymentsReconciler) clusterControllerLogs(clusterInstall *hiveext.AgentClusterInstall, cluster *common.Cluster) {
+	var status corev1.ConditionStatus
+	var reason, message string
+	switch {
+	case !swag.IsZero(cluster.ControllerLogsCollectedAt):
+		status = corev1.ConditionTrue
+		reason = LogCollectionCompletedReason
+		downloadURL := fmt.Sprintf("%s%s/clusters/%s/logs",
+			r.ServiceBaseURL, restclient.DefaultBasePath, cluster.ID.String())
+		message = fmt.Sprintf("%s - download URL: %s", LogCollectionCompletedMsg, downloadURL)
+	case cluster.LogsInfo == models.LogsStateTimeout:
+		status = corev1.ConditionFalse
+		reason = LogCollectionTimeoutReason
+		message = LogCollectionTimeoutMsg
+	case cluster.LogsInfo == models.LogsStateRequested:
+		status = corev1.ConditionFalse
+		reason = LogCollectionRequestedReason
+		message = LogCollectionRequestedMsg
+	case cluster.LogsInfo == models.LogsStateCollecting:
+		status = corev1.ConditionFalse
+		reason = LogCollectionCollectingReason
+		message = LogCollectionCollectingMsg
+	default:
+		status = corev1.ConditionFalse
+		reason = LogCollectionUnavailableReason
+		message = LogCollectionUnavailableMsg
+	}
+
+	setClusterCondition(&clusterInstall.Status.Conditions, hivev1.ClusterInstallCondition{
+		Type:    LogCollectionCompletedCondition,
```

Comment:
Not sure I understand why we need two conditions

Reply:
One for stop, second for completed - for two sections in hive code (we already talked about that, what am I missing?)

Reply:
Hive is not looking for Logs conditions. As per Avishai request we can leave it with two conditions to make it clear.